gooddata-pandas/gooddata_pandas/result_convertor.py

```diff
@@ -0,0 +1,309 @@
+
+        # get dimension indexes mapping from response like {"dim1": 0, "dim0": 1}
+        dim_idx_dict = {dim["localIdentifier"]: idx for idx, dim in enumerate(response.dimensions)}
+
+        for grand_total in grand_totals:
+            # 2-dim results have always 1-dim grand totals (3-dim results have 2-dim gt but DataFrame stores 2D only)
+            dims = grand_total["totalDimensions"]
+            assert len(dims) == 1, "Only 2-dimensional results are supported"
+            dim_idx = dim_idx_dict[dims[0]]
+            # the dimension id specified on the grand total says from what dimension were
+            # the grand totals calculated (1 for column totals or 0 for row totals);
+            #
+            # the grand totals themselves should, however be placed in the opposite dimension:
+            #
+            # column totals are extra rows at the end of the data
+            # row totals are extra columns at the right 'edge' of the data
+            opposite_dim = 1 if dim_idx == 0 else 0
+
+            if self.grand_totals[opposite_dim] is None:
+                # grand totals not initialized yet; initialize both data and headers by making
+                # a shallow copy from the results
+                self.grand_totals[opposite_dim] = grand_total["data"][:]
+                # TODO: row total measure headers are currently not supported (only aggregation info w/o measure label)
+                #       measure header defs are under ["headerGroups"][>0]
+                self.grand_totals_headers[opposite_dim] = grand_total["dimensionHeaders"][0]["headerGroups"][0][
+                    "headers"
+                ][:]
+            elif paging_dim != opposite_dim:
+                # grand totals are already initialized and the code is paging in the direction that reveals
+                # additional grand total values; append them accordingly; no need to consider total headers:
+                # that is because only the grand total data is subject to paging
+                grand_totals_item = cast(List[_DataArray], self.grand_totals[opposite_dim])
+                if opposite_dim == 0:
+                    # have column totals and paging 'to the right'; totals for the new columns are revealed so
+                    # extend existing data arrays
+                    for total_idx, total_data in enumerate(grand_total["data"]):
+                        grand_totals_item[total_idx].extend(total_data)
+                else:
+                    # have row totals and paging down, keep adding extra rows
+                    grand_totals_item.extend(grand_total["data"])
+
+    def result(self) -> _DataWithHeaders:
+        return _DataWithHeaders(
+            data=self.data,
+            data_headers=(cast(_DataHeaders, self.data_headers[0]), self.data_headers[1]),
+            grand_totals=(self.grand_totals[0], self.grand_totals[1]),
+            grand_total_headers=(self.grand_totals_headers[0], self.grand_totals_headers[1]),
```

Comment:
Can't we just do the following:

grand_total_headers=self.grand_total_headers?

If I understand correctly from the source code the self.grand_total_headers will always be of length 2 or am I wrong?

Reply:
dtto grand_total above